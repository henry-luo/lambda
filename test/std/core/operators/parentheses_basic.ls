// Test: Parentheses Basic Grouping
// Category: core/operators
// Type: positive
// Expected: 20

// Test parentheses override precedence: (2 + 3) * 4 = 5 * 4 = 20
(2 + 3) * 4
