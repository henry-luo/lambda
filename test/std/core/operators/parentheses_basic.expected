09:15:47 [NOTICE] Start transpiling test/std/core/operators/parentheses_basic.ls...
[script:int]
  [binary expr *.5:int]
    [primary expr:int,const:0]
      [binary expr +.3:int]
        [primary expr:int,const:1]
          [const: int]
        [primary expr:int,const:1]
          [const: int]
    [primary expr:int,const:1]
      [const: int]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/parentheses_basic.ls' to MIR
C2MIR init end           -- 74 usec
test/std/core/operators/parentheses_basic.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2398, parse tokens -- 1441
  C2MIR preprocessor end    -- 513 usec
  C2MIR parser end          -- 627 usec
  C2MIR context checker end -- 703 usec
  C2MIR generator end       -- 745 usec
C2MIR compiler end                -- 780 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/parentheses_basic.ls' executed: #####
20
